sound/soc/sof/amd/acp-ipc.c

```diff
@@ -200,6 +202,27 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
 		return IRQ_HANDLED;
 	}
 
+	if (desc->probe_reg_offset) {
+		u32 val;
+		u32 posn;
+
+		/* Probe register consists of two parts
+		 * (0-30) bit has cumulative position value
+		 * 31 bit is a synchronization flag between DSP and CPU
+		 * for the position update
+		 */
+		val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, desc->probe_reg_offset);
+		if (val & PROBE_STATUS_BIT) {
+			posn = val & ~PROBE_STATUS_BIT;
+			if (adata->probe_stream) {
+				adata->probe_stream->cstream_posn = posn;
```

Comment:
should there be something that takes care of wrap-around if you use this capability for a long time the 32-bit byte counter will overflow?

Reply:
About this, We discussed internally , we almost store 2 GB  For debugging it is ok, After this DSP will not send position update to host, Application won't read. There we can close the crecord command.

Reply:
ok, you probably want to add a comment here with the explanations above. Others will have the same comment when pushing this upstream.

Reply:
Done. Added comment and pushed.